### PR TITLE
Add support for multiple locales in locale_gen

### LINKED ISSUE #8677: locale_gen allow configuring multiple locales
### Summary

The [locale_gen](https://docs.ansible.com/ansible/latest/collections/community/general/locale_gen_module.html) module currently allows for a single locale to be configured at a time. Using a `loop` here isn't the most efficient because both the detection of the current state and the generation of the locale after change needs to be run for every locale. If there are no objections, I would like to add the ability to configure multiple at the same time.

Regarding implementation, to avoid breaking changes, I would add a names parameter and keep the existing one. Is the best practice for Ansible?

### Issue Type

Feature Idea

### Component Name

locale_gen

### Additional Information

```yaml
- name: Set and generate locale
  community.general.locale_gen:
    names:
      - de_CH.UTF-8
      - en_US.UTF-8
    state: present
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by MatthiasKunnen**
##### SUMMARY

locale_gen can now take multiple locales at the same time.
This should speed it up compared to using loop.

Fixes #8677 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
locale_gen

##### ADDITIONAL INFORMATION
This change is **not** breaking.
```yaml
- name: Ensure locales exist # Previous (still possible)
  community.general.locale_gen:
    name: "{{ item }}"
    state: present
  loop:
    - en_GB.UTF-8
    - nl_NL.UTF-8

- name: Ensure locales exist # Current
  community.general.locale_gen:
    name:
      - en_GB.UTF-8
      - nl_NL.UTF-8
    state: present
```

### CODE REVIEW

File: plugins/modules/locale_gen.py [hunk 1]
```diff
@@ -25,9 +25,10 @@
         support: none
 options:
     name:
-        type: str
+        type: list
+        elements: str
         description:
-            - Name and encoding of the locale, such as "en_GB.UTF-8".
+            - Name and encoding of the locales, such as "en_GB.UTF-8".
```
Comment (bcoca) [inline plugins/modules/locale_gen.py:31]: you should add note 'before version x.y.z this was a string'

Comment (MatthiasKunnen) [reply plugins/modules/locale_gen.py:31]: Added

**Comment by MatthiasKunnen** [comment]
The locale_gen module contains some special code for handling Ubuntu, however, on none of the test images, this code path is hit. The special path triggers when `/var/lib/locales/supported.d/` is present on the system. This is not the case for either `ubuntu2022` or `ubuntu2024`.

I can confirm that this directory is present on a ubuntu 2024 desktop installation. 

What would be the best way to test this? Testing on my ubuntu 2024 desktop installation would be possible but quite a chore and leaves the code untested in the CI.

**Review by felixfontein — COMMENTED** [review]
I didn't yet manage to fully review this, but here are some first comments.

File: plugins/modules/locale_gen.py [hunk 2]
```diff
@@ -25,9 +25,11 @@
...
```
[the remark this answers is not in the document]
Comment (MatthiasKunnen) [reply plugins/modules/locale_gen.py:31]: What does this `V` notation do? Could you link me to any relevant documentation?

[the remark this answers is not in the document]
Comment (felixfontein) [reply plugins/modules/locale_gen.py:31]: See here: https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#semantic-markup-within-module-documentation

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this on Monday.

**Comment by felixfontein** [comment]
@MatthiasKunnen thanks for your contribution!
@bcoca thanks for reviewing!

**Pull request merged**